src/Task/TaskFileSeeYou.cpp

```diff
@@ -378,20 +381,26 @@ CreateOZ(const SeeYouTurnpointInformation &turnpoint_infos,
 
     const Angle A12adj = CalcIntermediateAngle(turnpoint_infos,
                                                wp->location,
+                                               wps[0]->location,
                                                wps[pos - 1]->location,
-                                               wps[pos + 1]->location,
-                                               wps[0]->location);
+                                               wps[pos + 1]->location);
 
     const Angle RadialStart = (A12adj - turnpoint_infos.angle1).AsBearing();
     const Angle RadialEnd = (A12adj + turnpoint_infos.angle1).AsBearing();
 
     if (turnpoint_infos.radius2 > 0 &&
-        (turnpoint_infos.angle2.AsBearing().Degrees()) < 1) {
-      oz = new AnnularSectorZone(wp->location, turnpoint_infos.radius1,
-          RadialStart, RadialEnd, turnpoint_infos.radius2);
+        (turnpoint_infos.angle2.AsBearing().Degrees()) != 0) {
+      oz = VariableKeyholeZone::New(wp->location,
```

Comment:
This change needs some explanation. Maybe move the whole SeeYou parser change to a separate, properly documented commit. It's not obvious what this change is doing, why it modifies existing code. Maybe the old code was always wrong and this fixes a real problem - but it deserves an explanation.